core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -535,50 +554,50 @@ public void onEvent(RequestEvent event) {
           responseSize = 0;
         }
 
+        final MethodMetrics globalMetrics = Objects.requireNonNull(getGlobalMetrics(event));
         // Handle exceptions
         if (event.getException() != null) {
-          this.metrics.get(null).metrics().exception(event);
+          globalMetrics.exception(event);
+
+          // get RequestScopedMetrics for a single resource method
           final MethodMetrics metrics = getMethodMetrics(event);
           if (metrics != null) {
             metrics.exception(event);
           }
         }
 
-        this.metrics.get(null).metrics().finished(requestSize, responseSize, elapsed);
+        globalMetrics.finished(requestSize, responseSize, elapsed);
         final MethodMetrics metrics = getMethodMetrics(event);
         if (metrics != null) {
           metrics.finished(requestSize, responseSize, elapsed);
         }
       }
     }
 
+    private MethodMetrics getGlobalMetrics(RequestEvent event) {
+      // (null key) RequestScopedMetrics is the global metrics for ALL resource methods, see
+      // io.confluent.rest.metrics.MetricsResourceMethodApplicationListener.onEvent in this file
+      RequestScopedMetrics globalRequestScopedMetrics = this.metrics.get(null);
+      Object enabledGlobalStatsTasObj = event.getContainerRequest()
```

Comment:
`GLOBAL_STATS_REQUEST_TAGS_ENABLED_PROP_KEY` is property that should be set once when MetricsResourceMethodApplicationListener is created for the application. Why do you need to set it for each-request?

This should come in when the listener is constructed [here](https://github.com/RichRelevance/rest-utils/blob/f32e381e7af65f57edbe0234445b7a506189d6e4/core/src/main/java/io/confluent/rest/Application.java#L485). `GLOBAL_STATS_REQUEST_TAGS_ENABLED_PROP_KEY` could then be `RestConfig`

Reply:
@msn-tldr good catch, thanks